### ISSUE TITLE
check operator "date_greater_than" without component parameter

### ISSUE THREAD

**Issue by HajimeShimizu** [cdisc-org/cdisc-rules-engine#1525]
This is report for CORE v0.14.0.

I created the following rule.
```
Check:
  all:
   - name: --STDTC
     operator: date_greater_than
     value: --ENDTC
```
I prepared the 2 test cases:
 (1) --STDTC: 2025-01-10　--ENDTC: 2025
 (2) --STDTC: 2025-01-10　--ENDTC: 2025-01

Check reported error in case 2, but not for case 1.

I reviewed check_operations/helper.py and found following:

```
def _compare_with_inferred_precision(...):
    target_precision = context["target_precision"]
    comparator_precision = context["comparator_precision"]

　...(some codes)...

    result = operator_func(truncated_target, truncated_comparator)

    if truncated_target == truncated_comparator:
        if target_precision and comparator_precision:  # <--- something is wrong? 
            if target_precision.value > comparator_precision.value:
                return operator_func(get_date(target), get_date(comparator))
        return result

    return result
```

As far as I see, when the data is only "year" part, precision is set zero(0).
Hence, "target_precision and comparator_precision" equals False.

I believe this is not intended specification. If target_precision = 3 and comparator_precision = 0, additional check need to be done . 

```
    if truncated_target == truncated_comparator:
        if target_precision is not None and comparator_precision is not None:  # <--- we want to exclude None for this check 
            if target_precision.value > comparator_precision.value:
                return operator_func(get_date(target), get_date(comparator))
        return result
```
Is this what we want?
